core/storage/feedback/gae_models.py

```diff
@@ -73,9 +73,18 @@ class FeedbackThreadModel(base_models.BaseModel):
 
     @classmethod
     def generate_new_thread_id(cls, exploration_id):
-        """Generates a new thread id, unique within the exploration.
+        """Generates a new thread ID which is unique within the exploration.
 
-        Exploration ID + the generated thread ID is globally unique.
+        Args:
+            exploration_id: str, ID associated with each exploration
+
+        Returns:
+            thread_id: str, Thread ID is expressed as a sum of current time in
+            milliseconds and a random integer, or none if thread with same ID
```

Comment:
Alas, not quite. Let me explain what the method is doing.

This method is repeating the following loop N times: it generates a random thread_id, and checks if it exists. If it doesn't exist, the random thread_id that's just been generated is returned. Otherwise, it tries again. If, after N times, this generation does not work, it gives up and raises an exception.

An appropriate docstring for the "Returns" section is:

```
thread_id: str. A thread ID that is different from the IDs of all existing threads in the given exploration.
```

An appropriate docstring for the "Raises" section is:

```
Exception: There were too many collisions with existing thread IDs when attempting to generate a new thread ID.
```